### PR TITLE
[Flashcard] Delete flashcard

### LINKED ISSUE #138: [1.x] Delete a card
### Description

A user can delete a card.

**Acceptance Criteria**

- [x] A user can delete a card
- [x] Only the card's owner can delete the card
- [x] It's a soft delete

### Example

_No response_

### PR DESCRIPTION
**Pull request by imdhemy**
| Q             | A                                                                  |
|---------------|--------------------------------------------------------------------|
| Branch?       | 1.x                                              |
| Bug fix?      | no                                                             |
| New feature?  | yes                   |
| Deprecations? | no  |
| Tickets       | Fix #138             |
| License       | MIT                                                                |

> **Note**
> I don't like the way we test the actions, but let's improve them in a separate PR

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
   "require-dev": {
     "fakerphp/faker": "^1.20",
     "friendsofphp/php-cs-fixer": "3.13.2",
-    "phpunit/phpunit": "^9.6",
+    "phpunit/phpunit": "9.5.28",
```
Comment (imdhemy) [inline composer.json:37]: Reported in this issue https://github.com/piscibus/siklid-api/issues/228

**Comment by mgamal92** [comment]
@imdhemy there is a conflict in composer.lock.
BTW I'm reviewing this PR.

**Comment by imdhemy** [comment]
Nice! Let's finish this first #234

**Review by Moemen-Gaballah — COMMENTED** [review]
I think using Auth instead of **AuthChecker**  it's become better.

**Review by mgamal92 — APPROVED** [review]

**Review by Moemen-Gaballah — APPROVED** [review]
Good Job.

**Comment by mgamal92** [comment]
@imdhemy Please merge this PR as it blocks #173 as I need the updated version of the ```Request``` class to get variables from the bag

**Comment by imdhemy** [comment]
@Moemen-Gaballah 
> I think using Auth instead of AuthChecker it's become better.

Renamed 

@mgamal92 
> Please merge this PR as it blocks https://github.com/piscibus/siklid-api/pull/173 as I need the updated version of the Request class to get variables from the bag

Merged. As a side note for the future, you can cherry pick [the commit](https://github.com/piscibus/siklid-api/pull/227/commits/417c410878bac94d702d57d59c3926cae8aff5c4).

**Pull request merged**